特定画廊解析种子文件列表失败

### 复现步骤 / Step to reproduce

1. 进入 https://exhentai.org/g/2692070/7e2c468c66/ 
2. 进入种子下载页面

### 预期行为 / Expected behaviour

出现种子列表。

### 实际行为 / Actual Behaviour

出现“奇怪的错误”。

### 应用日志 / App logs

```
--------- beginning of main
10-22 23:16:02.140 15060 15060 W moe.tarsin.ehviewer: Failed to determine odex file name: Dex location /gmscompat_fd_120 has no extension.
10-22 23:16:02.140 15060 15060 W ziparchive: Unable to open '/gmscompat_fd_120.dm': No such file or directory
10-22 23:16:02.150 15060 15060 I DynamiteModule: Considering local module com.google.android.gms.cronet_dynamite:0 and remote module com.google.android.gms.cronet_dynamite:2411
10-22 23:16:02.150 15060 15060 I DynamiteModule: Selected remote version of com.google.android.gms.cronet_dynamite, version >= 2411
10-22 23:16:02.150 15060 15060 V DynamiteModule: Dynamite loader version >= 2, using loadModule2NoCrashUtils
10-22 23:16:02.163 15060 15060 W System  : ClassLoader referenced unknown path: 
10-22 23:16:02.165 15060 15060 D nativeloader: Configuring clns-4 for other apk . target_sdk_version=34, uses_libraries=, library_path=/data/app/~~ZjXJajw4wo86TGCgESxmtQ==/com.google.android.gms-8KCRLrNCj6Yo8UQJ64EPtw==/lib/arm64:/data/app/~~ZjXJajw4wo86TGCgESxmtQ==/com.google.android.gms-8KCRLrNCj6Yo8UQJ64EPtw==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand
10-22 23:16:02.184 15060 15060 W moe.tarsin.ehviewer: Failed to determine odex file name: Dex location /gmscompat_fd_150 has no extension.
10-22 23:16:02.184 15060 15060 W ziparchive: Unable to open '/gmscompat_fd_150.dm': No such file or directory
10-22 23:16:02.188 15060 15060 I moe.tarsin.ehviewer: Waiting for a blocking GC ClassLinker
10-22 23:16:02.244 15060 15060 I moe.tarsin.ehviewer: WaitForGcToComplete blocked ClassLinker on Background for 55.726ms
10-22 23:16:02.260 15060 15084 W System  : A resource failed to call close. 
10-22 23:16:02.378 15060 15096 D nativeloader: Configuring clns-5 for other apk . target_sdk_version=34, uses_libraries=ALL, library_path=/gmscompat_fd_150!/lib/arm64-v8a, permitted_path=/data:/mnt/expand
10-22 23:16:02.379 15060 15096 W linker  : Warning: unable to resolve "/gmscompat_fd_150": No such file or directory (ignoring)
10-22 23:16:02.380 15060 15060 W DefaultDispatch: type=1400 audit(0.0:130421): avc:  granted  { execute } for  path="/data/user_de/0/com.google.android.gms/app_chimera/m/0000005a/CronetDynamite.apk" dev="dm-40" ino=320454 scontext=u:r:untrusted_app:s0:c201,c256,c512,c768 tcontext=u:object_r:app_data_file:s0:c158,c256,c512,c768 tclass=file app=moe.tarsin.ehviewer
10-22 23:16:02.394 15060 15096 I cr_CronetLibraryLoader: Cronet version: 114.0.5735.33, arch: aarch64
10-22 23:16:02.405 15060 15142 D TrafficStats: tagSocket(112) with statsTag=0xffffffff, statsUid=-1
10-22 23:16:02.422 15060 15142 D TrafficStats: tagSocket(168) with statsTag=0xffffffff, statsUid=-1
10-22 23:16:02.436 15060 15142 D TrafficStats: tagSocket(117) with statsTag=0xffffffff, statsUid=-1
10-22 23:16:02.438 15060 15142 D TrafficStats: tagSocket(117) with statsTag=0xffffffff, statsUid=-1
10-22 23:16:03.070 15060 15153 D ProfileInstaller: Installing profile for moe.tarsin.ehviewer
10-22 23:16:14.467 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:16:15.818 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:16:15.818 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:16:15.820 15060 15091 D EhEngine: https://exhentai.org/g/2692070/7e2c468c66/
10-22 23:16:15.821 15060 15060 I ImeTracker: moe.tarsin.ehviewer:5c8a5e4d: onRequestHide at ORIGIN_CLIENT_HIDE_SOFT_INPUT reason HIDE_SOFT_INPUT_BY_INSETS_API
10-22 23:16:15.822 15060 15060 I ImeTracker: moe.tarsin.ehviewer:5c8a5e4d: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-22 23:16:20.133 15060 15060 I AppCompatDelegate: The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
10-22 23:16:20.142 15060 15118 W Parcel  : Expecting binder but got null!
10-22 23:16:20.161 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: Exception in CalledByNative method
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: java.nio.BufferOverflowException
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at java.nio.ByteBuffer.put(ByteBuffer.java:656)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at java.nio.DirectByteBuffer.put(DirectByteBuffer.java:272)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at BF.a(Unknown Source:16)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at BF.q(Unknown Source:9)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at ou.onReadCompleted(Unknown Source:24)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at m.jf.onReadCompleted(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):3)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):34)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
10-22 23:16:22.290 15060 15123 E cr_CronetUrlRequestContext: 	at java.lang.Thread.run(Thread.java:1012)
10-22 23:16:22.293 15060 15060 W System.err: m.hf: Exception received from UrlRequest.Callback
10-22 23:16:22.293 15060 15060 W System.err: 	at org.chromium.net.impl.CronetUrlRequest.e(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):5)
10-22 23:16:22.293 15060 15060 W System.err: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):46)
10-22 23:16:22.293 15060 15060 W System.err: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
10-22 23:16:22.293 15060 15060 W System.err: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
10-22 23:16:22.293 15060 15060 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
10-22 23:16:22.293 15060 15060 W System.err: Caused by: java.nio.BufferOverflowException
10-22 23:16:22.293 15060 15060 W System.err: 	at java.nio.ByteBuffer.put(ByteBuffer.java:656)
10-22 23:16:22.293 15060 15060 W System.err: 	at java.nio.DirectByteBuffer.put(DirectByteBuffer.java:272)
10-22 23:16:22.293 15060 15060 W System.err: 	at BF.a(Unknown Source:16)
10-22 23:16:22.293 15060 15060 W System.err: 	at BF.q(Unknown Source:9)
10-22 23:16:22.293 15060 15060 W System.err: 	at ou.onReadCompleted(Unknown Source:24)
10-22 23:16:22.293 15060 15060 W System.err: 	at m.jf.onReadCompleted(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):3)
10-22 23:16:22.293 15060 15060 W System.err: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):34)
10-22 23:16:22.293 15060 15060 W System.err: 	... 3 more
10-22 23:16:22.318 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:16:23.064 15060 15060 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@6c42697
10-22 23:16:23.084 15060 15060 W InputEventReceiver: Attempted to finish an input event but the input event receiver has already been disposed.
10-22 23:16:49.168 15060 15142 D TrafficStats: tagSocket(116) with statsTag=0xffffffff, statsUid=-1
10-22 23:17:11.908 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:17:11.908 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:17:12.000 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:17:12.842 15060 15060 I AppCompatDelegate: The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
10-22 23:17:12.856 15060 15117 W Parcel  : Expecting binder but got null!
10-22 23:17:12.878 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: Exception in CalledByNative method
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: java.nio.BufferOverflowException
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at java.nio.ByteBuffer.put(ByteBuffer.java:656)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at java.nio.DirectByteBuffer.put(DirectByteBuffer.java:272)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at BF.a(Unknown Source:16)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at BF.q(Unknown Source:9)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at ou.onReadCompleted(Unknown Source:24)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at m.jf.onReadCompleted(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):3)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):34)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
10-22 23:17:14.993 15060 15123 E cr_CronetUrlRequestContext: 	at java.lang.Thread.run(Thread.java:1012)
10-22 23:17:14.998 15060 15060 W System.err: m.hf: Exception received from UrlRequest.Callback
10-22 23:17:14.998 15060 15060 W System.err: 	at org.chromium.net.impl.CronetUrlRequest.e(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):5)
10-22 23:17:14.998 15060 15060 W System.err: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):46)
10-22 23:17:14.998 15060 15060 W System.err: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
10-22 23:17:14.998 15060 15060 W System.err: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
10-22 23:17:14.998 15060 15060 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
10-22 23:17:14.998 15060 15060 W System.err: Caused by: java.nio.BufferOverflowException
10-22 23:17:14.998 15060 15060 W System.err: 	at java.nio.ByteBuffer.put(ByteBuffer.java:656)
10-22 23:17:14.998 15060 15060 W System.err: 	at java.nio.DirectByteBuffer.put(DirectByteBuffer.java:272)
10-22 23:17:14.998 15060 15060 W System.err: 	at BF.a(Unknown Source:16)
10-22 23:17:14.998 15060 15060 W System.err: 	at BF.q(Unknown Source:9)
10-22 23:17:14.998 15060 15060 W System.err: 	at ou.onReadCompleted(Unknown Source:24)
10-22 23:17:14.998 15060 15060 W System.err: 	at m.jf.onReadCompleted(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):3)
10-22 23:17:14.998 15060 15060 W System.err: 	at m.ik.run(:com.google.android.gms.dynamite_cronetdynamite@233717044@23.37.17 (190400-0):34)
10-22 23:17:14.998 15060 15060 W System.err: 	... 3 more
10-22 23:17:15.024 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:17:15.467 15060 15060 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@9049d27
10-22 23:17:15.484 15060 15060 W InputEventReceiver: Attempted to finish an input event but the input event receiver has already been disposed.
10-22 23:17:57.297 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:17:57.297 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:17:57.329 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:17:59.278 15060 15060 I ImeTracker: moe.tarsin.ehviewer:8087c976: onRequestHide at ORIGIN_CLIENT_HIDE_SOFT_INPUT reason HIDE_SOFT_INPUT_BY_INSETS_API
10-22 23:17:59.278 15060 15060 I ImeTracker: moe.tarsin.ehviewer:8087c976: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-22 23:17:59.726 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:17:59.726 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:18:00.119 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:18:00.119 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:18:00.404 15060 15060 I AppCompatDelegate: The Activity's LayoutInflater already has a Factory installed so we can not install AppCompat's
10-22 23:18:00.425 15060 15117 W Parcel  : Expecting binder but got null!
10-22 23:18:00.444 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:18:00.638 15060 15098 D OpenGLRenderer: endAllActiveAnimators on 0xb400d2c671f0a600 (NavigationMenuItemView) with handle 0xb400d197a17eef60
10-22 23:18:00.951 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:18:00.951 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:18:04.368 15060 15098 D OpenGLRenderer: endAllActiveAnimators on 0xb400d2c671f37000 (g01) with handle 0xb400d197a16f07b0
10-22 23:18:04.782 15060 15060 D CompatibilityChangeReporter: Compat change id reported: 78294732; UID 10201; state: ENABLED
10-22 23:18:04.782 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:18:04.782 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:18:04.793 15060 15060 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-22 23:18:04.793 15060 15060 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
10-22 23:18:04.794 15060 15117 W Parcel  : Expecting binder but got null!
10-22 23:18:04.812 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
10-22 23:18:04.833 15060 15098 E OpenGLRenderer: Unable to match the desired swap behavior.
```

### 备注 / Addition details

这个画廊与其他可以正常下载种子的画廊的区别在于，它有多个被 Expunged 的种子。目前猜测是像这个画廊一样的种子里面有被 Expunged 的的，解析不了。

### EhViewer 版本号 / EhViewer version code

1.8.10.1-gms

### EhViewer CI 版本 / EhViewer CI Version

6e2a2de

### Android 系统版本 / Android version

Android 14

### 设备型号 / Device model

N/A

### SoC 型号 / Soc model

N/A

### 自查步骤 / Verify steps

- [ ] 如果您有足够的时间和能力，并愿意为此提交 PR ，请勾上此复选框 / Pull request is welcome. Check this if you want to start a pull request
- [X] 您已搜索过 [Issue Tracker](https://github.com/FooIbar/EhViewer/issues)，没有找到类似内容 / I have searched on [Issue Tracker](https://github.com/FooIbar/EhViewer/issues), No duplicate or related open issue has been found
- [X] 您确保这个 Issue 只提及一个问题。如果您有多个问题报告，烦请发起多个 Issue / Ensure there is only one bug report in this issue. Please make mutiply issue for mutiply bugs
- [X] 您确保已使用最新 [CI 版本](https://github.com/FooIbar/EhViewer/actions)测试，并且该问题在最新 CI 版本中并未解决 / This bug have not solved in latest [CI version](https://github.com/FooIbar/EhViewer/actions)

--- a/app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt
+++ b/app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt
@@ -116,7 +116,7 @@ private fun rethrowExactly(code: Int, body: String, e: Throwable): Nothing {
     throw e
 }
 
-val httpContentPool = DirectByteBufferPool(16, 16384)
+val httpContentPool = DirectByteBufferPool(16, 0x20000)
 
 suspend inline fun <T> fetchCompat(url: String, referer: String? = null, crossinline parser: (ByteBuffer) -> T): T {
     return if (isCronetSupported) {

--- a/app/src/main/kotlin/com/hippo/ehviewer/util/ExceptionUtils.kt
+++ b/app/src/main/kotlin/com/hippo/ehviewer/util/ExceptionUtils.kt
@@ -16,7 +16,6 @@
 package com.hippo.ehviewer.util
 
 import com.hippo.ehviewer.R
-import com.hippo.ehviewer.client.exception.EhException
 import java.io.IOException
 import java.net.MalformedURLException
 import java.net.ProtocolException
@@ -60,8 +59,7 @@ object ExceptionUtils {
             }
 
             is SocketException, is SSLException -> appCtx.getString(R.string.error_socket)
-            is EhException -> e.message!!
-            else -> appCtx.getString(R.string.error_unknown)
+            else -> e.message ?: appCtx.getString(R.string.error_unknown)
         }
     }
 

